CHANGELOG.md

```diff
@@ -4,6 +4,243 @@ All notable changes to Router will be documented in this file.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/3562
+
+### Fix the error count for subscription requests for apollo telemetry ([PR #3500](https://github.com/apollographql/router/pull/3500))
+
+Count subscription requests only if the feature is enabled.
+
+The router would previously count subscription requests regardless of whether the feature is enabled or not. This changeset will only count subscription requests if the feature has been enabled.
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/3500
+
+## 🛠 Maintenance
+
+### Update datadog-subgraph npm dependencies ([PR #3560](https://github.com/apollographql/router/pull/3560))
+
+This changeset updates the dd-trace dependency and the nodeJS version of the example Dockerfile.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/3560
+
+### Remove some panic! calls from the pq code. ([PR #3527](https://github.com/apollographql/router/pull/3527))
+
+Replace a few `panic!` calls with `expect()` in the persisted query code for code clarity.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3527
+
+### Add a warning if we think istio-proxy injection is causing problems ([Issue #3533](https://github.com/apollographql/router/issues/3533))
+
+We have encountered situations where the injection of istio-proxy in a router pod (executing in Kubernetes) causes networking errors during uplink retrieval.
+
+The root cause is that the router is executing and attempting to retrieve uplink schemas while the istio-proxy is simultaneously modifying network configuration.
+
+This new warning message directs users to information which should help them to configure their Kubernetes cluster or pod to avoid this problem.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3545
+
+### Add a message to the logs indicating when custom plugins are detected and there is a possibility that log entries may be silenced ([Issue #3526](https://github.com/apollographql/router/issues/3526))
+
+Since [#3477](https://github.com/apollographql/router/pull/3477), users who have created custom plugins no longer see their log entries.
+This is because the default logging filter now restricts log entries to those that are in the apollo module.
+
+Users that have custom plugins will need to configure the logging filter to include their modules, but they may not realise this.
+
+Now, if a custom plugin is detected then a message will be logged to the console indicating that the logging filter may need to be configured.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3540
+
```

Comment:
Also a really nice user-debuggability win, @BrynCooke.  Love it. (Shortened the title)

```suggestion
### Log when custom plugins are detected with potentially-silenced log entries ([Issue #3526](https://github.com/apollographql/router/issues/3526))

Since [PR #3477](https://github.com/apollographql/router/pull/3477), users with custom plugins lost some log entries.  This is because the default logging filter now restricts log entries to those that are in the `apollo` module, as originally intended.

Users that have custom plugins need to configure the logging filter to include their modules, but may not have realised this.

Now, if a custom plugin is detected, a message will be logged to the console indicating that the logging filter may need to be configured.

By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3540

```